app/src/main/java/com/github/se/eduverse/ui/profile/FollowersListScreen.kt

```diff
@@ -87,39 +93,30 @@ fun FollowListScreen(
                         onFollowClick = { profileId ->
                           viewModel.toggleFollow(currentUserId ?: "", profileId)
                         },
-                        onProfileClick = { navigationActions.navigateToUserProfile(profile.id) })
+                        onProfileClick = { navigationActions.navigateToUserProfile(profile.id) },
+                        followActionState = followActionState,
+                        targetUserId =
+                            (followActionState as? FollowActionState.Success)?.targetUserId)
                   }
                 }
           }
         }
       }
 }
 
-// Extracted helper function for loading profiles
-private suspend fun loadProfiles(
-    viewModel: ProfileViewModel,
-    userId: String,
-    isFollowersList: Boolean,
-    onResult: (List<Profile>) -> Unit
-) {
-  val newProfiles =
-      if (isFollowersList) {
-        viewModel.getFollowers(userId)
-      } else {
-        viewModel.getFollowing(userId).map { it.copy(isFollowedByCurrentUser = true) }
-      }
-  onResult(newProfiles)
-}
-
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 private fun FollowListItem(
     profile: Profile,
     currentUserId: String?,
     isFollowersList: Boolean,
     onFollowClick: (String) -> Unit,
-    onProfileClick: () -> Unit
+    onProfileClick: () -> Unit,
+    followActionState: FollowActionState,
+    targetUserId: String?
 ) {
+  val isLoading = followActionState is FollowActionState.Loading && targetUserId == profile.id
+
```

Comment:
I feel like this will allways be false, because if followActionState if FollowActionState.Loading then the conversion to Success in call (l.99) will fail and return null, so targetUserId will be null